inc/input_queue.hpp

```diff
@@ -3,38 +3,68 @@
 
 namespace dragonslave {
 
-
 enum class InputType {
     KEYBOARD,
     MOUSE_POS,
     MOUSE_BUTTON,
     MOUSE_SCROLL,
 };
 
-
-struct InputEvent 
+struct MouseEvent
 {
-    InputType type;
-    int action = 0;
-    char key = 0;
     double x = 0, y = 0;
+    int button = 0;
```

Comment:
Would this be three considering there's mouse press, mouse position, and mouse scroll?